sensing/src/main/java/com/google/android/sensing/db/impl/DatabaseImpl.kt

```diff
@@ -78,6 +79,33 @@ internal class DatabaseImpl(context: Context, databaseConfig: DatabaseConfig) :
     return resourceInfoDao.getResourceInfo(resourceInfoId)
   }
 
+  override suspend fun getCaptureInfo(captureId: String): CaptureInfo {
+    return captureInfoDao.getCaptureInfo(captureId)
+      ?: throw ResourceNotFoundException("CaptureInfo", captureId)
+  }
+
+  override suspend fun deleteCaptureInfo(captureId: String): Boolean {
+    return captureInfoDao.deleteCaptureInfo(captureId) == 1
+  }
+
+  override suspend fun deleteResourceInfo(resourceInfoId: String): Boolean {
+    return resourceInfoDao.deleteResourceInfo(resourceInfoId) == 1
+  }
+
+  override suspend fun deleteUploadRequest(resourceInfoId: String): Boolean {
+    return uploadRequestDao.deleteUploadRequest(resourceInfoId) == 1
+  }
+
+  override suspend fun deleteRecordsInCapture(captureId: String): Boolean {
+    var deleted = true
+    listResourceInfoInCapture(captureId).forEach {
```

Comment:
Since DeleteOnCascade is enabled, deleting captureInfo alone should auto delete the child records. Can you please check this once?